src/components/fields/DataSelector.js

```diff
@@ -44,7 +44,16 @@ class DataSelector extends Component {
     }
     const update = {};
     if (this.dataSrcExists) {
-      update[this.srcAttr] = value;
+      const srcAttr = this.srcAttr;
+      update[srcAttr] = value;
+
+      if (!value) {
+        /*
+         * we're deleting a data attribute, let's also clean it up from gd.data
+         * to prevent issues when switching plots (i.e. candlestick -> box plot)
+         */
+        update[srcAttr.replace('src', '')] = value;
```

Comment:
so value here is supposed to `undefined`? Should we make that more clear? What if `value=0` or `value=''` or other falsey values?

Reply:
value gets set to null (from react-select) when you clear the dropdown (reset it completely)
and that is the situation that we're interested here, because when you clear a data field, you don't want gd.data to show x [ ], you'd prefer for gd.data to not have x at all, 

This is helpful for situations where you changed from one trace type that had x as a possible attribute to another trace that has that attribute, (candlestick -> box) but it's used in another way and so it can break the plot

Reply:
I updated the comment